### ISSUE TITLE
Keybindings: Add Keyboard Shortcuts to nteract web

### ISSUE THREAD

**Issue by stormpython** [nteract/archived-desktop-app#4101]
**Is your feature request related to a problem? Please describe.**
In order to bring `nteract` up to par with `Jupyter Classic`, we need to add keyboard shortcuts. I am closing issue #1870 in favor of this one.

**Describe the solution you'd like**
This will be a parent issue that has several child issues. This will require setting up a process for adding keyboard shortcuts, which can be hard-coded initially with the end goal being to have a keyboard shortcut config file that is eventually configurable in the UI. Once a system is in place, each class of keyboard shortcuts can be broken out and tackled in their own separate issues.

To Do List:
- [x] [#4102] Create system for adding keyboard shortcuts
~(**#4102 needs to be completed first, before the remaining issues can be started**)~
- [ ] [#4103] Add edit shortcuts
- [x] [#4104] Add file shortcuts
- [x] [#4105] Add notebook shortcuts
- [x] [#4106] Add kernel shortcuts
- [ ] [#4107] Create a keyboard shortcut modal

**Describe alternatives you've considered**
Tried using `blueprintjs` `Hotkey` component. I ran into an issue with the `HotkeysTarget` decorator component. The first error I was getting was not being able to use the decorator because it was expecting to call the wrapped component with `new`. 

```typescript
`Uncaught TypeError Class constructor App cannot be invoked without 'new'`
```

This still seems to be an issue, but I've been looking at this [issue](https://github.com/palantir/blueprint/issues/925) to help solve it. (**Workaround did not work**)

I also had an issue getting the event listeners to fire for `keydown` events. Kyle removed hot module loading to see if it had anything to do with the build, but I am still experiencing issues with getting the event listeners to fire. Turns out, there is an open [issue](https://github.com/palantir/blueprint/issues/1725) for `Hotkeys` not working on anything besides a `DOM` element.

**Comment by rgbkrk** [comment]
> Turns out, there is an open issue for Hotkeys not working on anything besides a DOM element.

Oh that makes sense. We'd definitely need a wrapper div. What did you think about the _other_ [react-hotkeys](https://github.com/greena13/react-hotkeys)?

**Comment by rgbkrk** [comment]
I just noticed that `react-hotkeys` (only one dash) uses [mousetrap](https://craig.is/killing/mice) underneath, which also allows key sequences. That was the one @theengineear originally was recommending we start with back in the day.

**Comment by stormpython** [comment]
Ok, let me take another look.

**Comment by stormpython** [comment]
@rgbkrk you should take a look at this [issue](https://github.com/greena13/react-hotkeys/issues/130). He mentions wanting to refactor `react-hotkeys` to remove **mousetrap** because it is no longer being maintained. Do you know if that is true?

**Comment by rgbkrk** [comment]
Omg wow what a good find. Reading now.

**Comment by rgbkrk** [comment]
Couldn't help myself and made a PR to the project. It looks like the maintainer felt burnt out then came back to the project. https://github.com/greena13/react-hotkeys/pull/139

If you like the project, we could always offer to host it in the nteract org and maintain it if you like the way it's written. Mousetrap does make new releases it looks like. Perhaps there are outstanding issues folks want fixed?

**Comment by stormpython** [comment]
@rgbkrk Yes, on further review, it seems to be the case. Let me dig around more in the code and get back to you.

**Comment by stormpython** [comment]
Ok, so the path forward now is to use the [React HotKeys](https://github.com/greena13/react-hotkeys) which uses [mousetrap.js](https://craig.is/killing/mice) under the hood. This library was chosen because it wraps `mousetrapjs` and it allows for key combinations, sequences, etc. The only slight drawback may be that the library may not be maintained. See this [issue](https://github.com/greena13/react-hotkeys/issues/130).

**Comment by captainsafia** [comment]
Closing in favor of https://github.com/nteract/nteract/issues/4781.
